src/diff/mount.js

```diff
@@ -240,16 +240,13 @@ export function mountChildren(
 		// Terser removes the `continue` here and wraps the loop body
 		// in a `if (childVNode) { ... } condition
 		if (childVNode == null) {
+			parentInternal._children[i] = null;
```

Comment:
+1 Noticed that too that we were leaving a hole here when stepping through our diff.